### PR TITLE
documentation: remove release docs

### PR DESCRIPTION
**Pull request by aanm**
The release process is now part of github.com/cilium/release and all of the steps are described in there. Thus, we can remove this part of the documentation to avoid confusion.

### CODE REVIEW

**Review by bimmlerd — CHANGES_REQUESTED** [review]
I don't see a link to `cilium/release`. Should we not be linking to that somewhere?

File: Documentation/contributing/release/index.rst [hunk 1]
```diff
@@ -19,9 +19,6 @@ upcoming stable releases.
 
    organization
    backports
-   stable
-   rc
-   feature
```
Comment (qmonnet) [inline Documentation/contributing/release/index.rst:24]: Please update the paragraph above (`This includes information around the release cycles and guides for developers responsible for [...] preparing upcoming stable releases`). As suggested by David, a pointer to the `release` repo would be helpful.

**Review by qmonnet — CHANGES_REQUESTED** [review]

File: Documentation/contributing/release/index.rst (hunk 1 above)
Comment (aanm) [reply Documentation/contributing/release/index.rst:24]: Ups, good point!

File: Documentation/contributing/release/index.rst [hunk 2]
```diff
@@ -9,19 +9,16 @@
...
 ------------------
 
-This section describes the release processes for tracking, preparing, and
-creating new Cilium releases. This includes information around the release
-cycles and guides for developers responsible for backporting fixes or preparing
-upcoming stable releases.
+This section includes information around the release cycles and guides for
+developers responsible for backporting fixes.
+
+For release preparation the steps can be found under https://github.com/cilium/release/issues/new/choose
```
Comment (bimmlerd) [inline Documentation/contributing/release/index.rst]: nit, something like

```suggestion
Release preparation steps can be found under https://github.com/cilium/release/issues/new/choose. 
```

**Review by bimmlerd — APPROVED** [review]

**Review by michi-covalent — APPROVED** [review]
🧹🧹🧹

**Review by qmonnet — APPROVED** [review]

**Comment by qmonnet** [comment]
Travis failure seems to be the Travis flavour of https://github.com/cilium/cilium/issues/24087.

**Pull request merged**